### PR TITLE
Navigator: land: fix when flying without global position estimate

### PR DESCRIPTION
**Pull request by sfuhrer**
### Solved Problem
Fixes https://github.com/PX4/PX4-Autopilot/issues/21524
Fixes https://github.com/PX4/PX4-Autopilot/issues/23773
Proposes a partial alternative to https://github.com/PX4/PX4-Autopilot/pull/23839

Land without global position estimate (e.g. GNSS-denied without manually initialized position) was broken since https://github.com/PX4/PX4-Autopilot/pull/21036/commits/efc8c167acd57ca175b9f2eea795b750227263b3. Instead of landing at the current spot it flew to (0/0).

### Solution
Set lat/lon fields of triplet to NAN if global position is not valid. Further I disable the [breaking distance calculation](https://github.com/PX4/PX4-Autopilot/pull/23546) in this case, as it's requiring global position. Will be anyway removed from Navigator with https://github.com/PX4/PX4-Autopilot/pull/23678.

### Changelog Entry
For release notes:
```
Bugfix: Navigator: land: fix when flying without global position estimate
```

### Test coverage
SITL tested.

### Context
Land without global position only works with MC vehicles, FW requires global position.

### CODE REVIEW

**Review by AlexisTM — APPROVED** [review]
I tested it in simulation and it properly lands in position without a GPS

**Comment by ptrindade96** [comment]
Hi @sfuhrer. It would be nice if you could backport this to v1.14, since the same issue occurs with that version.

You can take a look in [this discussion](https://discuss.px4.io/t/auto-land-going-to-the-origin-with-optical-flow/40767).

File: src/modules/navigator/land.cpp [hunk 1]
```diff
@@ -58,7 +58,8 @@ Land::on_activation()
 	/* convert mission item to current setpoint */
 	struct position_setpoint_triplet_s *pos_sp_triplet = _navigator->get_position_setpoint_triplet();
 
-	if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING) {
+	if (_navigator->get_vstatus()->vehicle_type == vehicle_status_s::VEHICLE_TYPE_ROTARY_WING
+	    && _navigator->get_local_position()->xy_global) { // only execute if global position is valid
```
Comment (MaEtUgR) [inline src/modules/navigator/land.cpp:62]: This should also be done further down in the file after the VTOL transition then.

Comment (MaEtUgR) [reply src/modules/navigator/land.cpp:62]: Done in ea98e1d

**Review by MaEtUgR — APPROVED** [review]
Sorry I blew up your pr with the last naming refactoring commit but the breaking functions haunted me 😇 Feel free to change it again or I can also make a separate pr.

**Comment by MaEtUgR** [comment]
> It would be nice if you could backport this to v1.14, since the same issue occurs with that version.

We usually only do point releases on the last stable but it's maybe worth a backport if people are struggling...

File: src/modules/navigator/land.cpp (hunk 1 above)
Comment (sfuhrer) [reply src/modules/navigator/land.cpp:62]: Yes correct, though a VTOL couldn't fly with local position only in fixed-wing in the first place. Still, doesn't hurt to add this guard.

**Comment by sfuhrer** [comment]
> Hi @sfuhrer. It would be nice if you could backport this to v1.14, since the same issue occurs with that version.
> 
> You can take a look in [this discussion](https://discuss.px4.io/t/auto-land-going-to-the-origin-with-optical-flow/40767).

I don't think there are any more 1.14 releases planned, as we now have 1.15 out already. I will thus port it to the 1.15 branch.

**Comment by duguguang** [comment]
how about the takeoff command?

**Comment by ptrindade96** [comment]
> > Hi @sfuhrer. It would be nice if you could backport this to v1.14, since the same issue occurs with that version.
> > You can take a look in [this discussion](https://discuss.px4.io/t/auto-land-going-to-the-origin-with-optical-flow/40767).
> 
> I don't think there are any more 1.14 releases planned, as we now have 1.15 out already. I will thus port it to the 1.15 branch.

That's ok, at least for me it makes no difference. Thanks for your help Silvan.

**Comment by sfuhrer** [comment]
> how about the takeoff command?

Good point. I've re-opened https://github.com/PX4/PX4-Autopilot/issues/21524.

**Pull request merged**